### ISSUE TITLE
Protobuf deserialization has a default of an empty String

### ISSUE THREAD

**Issue by ahus1** [keycloak/keycloak#30761]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Acceptance criteria

* Adding toggling the new behavior in protostream library
* Protostream release with the new feature
* Adapting the Keycloak code base to use the toggle
* Consider a soon-ish productized version of protostream (make it part of this issue, or create a new issue for KC26)

### Describe the bug

In protocol buffers v3 the default value of a String is an empty string. Due to that in the current implementation, all properties of a serialized objected with are null strings will become an empty string.

### Version

main

### Regression

- [X] The issue is a regression

### Expected behavior

A null String field should still be null after serialization and de-serialization. 

### Actual behavior

The value will become an empty String instead. 

In commits 66dd9e65b97b06321b3d07856e1bc56c267d1184 and 67098f04694cf0e06a4bd25c477b5110313a7228, this has already been fixed by treating any empty string as a null value. 

We are still missing it in other places like for example the brokerSessionId in the UserSessionEntity. 

### Anything else?

In a team discussion on Jun 25 I raised my concern that the test suite of Keycloak is not prepared to find those test failures as they will appear only when using the new external Infinispan setup where values are always serialized, or in a setup with multiple Keycloak nodes where keys are read from and written to another node. 

To keep the existing behavior of `MarshallUtil.marshallString` which kept null values, we were discussed with @pruivo, @ryanemerson, @mhajas and @kami619 to ask for a similar behavior in protostream. Ideally we would like to have a central switch to have all String values treated as null by default. 

Once the change is done, `Marshalling.emptyStringToNull` should no longer be necessary and can therefore be removed.

**Comment by pruivo** [comment]
Created https://issues.redhat.com/browse/IPROTO-343

**Comment by pruivo** [comment]
A little bit of reading for those interested: https://github.com/google/guava/wiki/UsingAndAvoidingNullExplained
